src/runtime/virtcontainers/factory/template/template_test.go

```diff
@@ -129,7 +129,18 @@ func TestTemplateFactory(t *testing.T) {
 	// CloseFactory, there is no need to call tt.CloseFactory(ctx)
 	f.CloseFactory(ctx)
 
-	// expect tt.statePath not exist, if exist, it means this case failed.
-	_, err = os.Stat(tt.statePath)
-	assert.Nil(err)
+	//umount may take more time. Check periodically if the mount exists
+	check := 20
+	delay := 1 * time.Second
+	for check > 0 {
+		// expect tt.statePath not exist, if exist, it means this case failed.
+		_, err = os.Stat(tt.statePath)
+		if err != nil {
+			break
+		} else {
+			check -= 1
+			time.Sleep(delay)
+		}
+	}
+	assert.True(os.IsNotExist(err))
```

Comment:
Can you change this call to give more details on failure. Something like:

```suggestion
	assert.True(os.IsNotExist(err), fmt.Sprintf("mount still present after waiting %d seconds", check))
```

Of course, that won't actually work as you are decrementing `check` currently.

Reply:
yes, adding this might be confusing since the mounts would always be removed in this case.

Reply:
The message will only be displayed if the assertion fails so it might be worth adding it but adding a new variable so you can show how long the test waited before failing.

Reply:
@jodh-intel , updated. PTAL